app/models/subscriber.rb

```diff
@@ -0,0 +1,32 @@
+class Subscriber
+  def self.for_discussion(discussion)
+    new(discussion)
+  end
+
+  def initialize(discussion)
+    @discussion = discussion
+    @subscriptions = Subscription.where(discussion_id: discussion.id)
+  end
```

Comment:
Ok, makes sense

Just as a sidenote though, I don't see why you couldn't do something similar to the swift syntax:

```ruby
def initialize(subscribed_discussion:)
  @discussion = subscribed_discussion)
end
```

Also, making the initialize private doesn't do anything (you can still instantiate with `new`), unless it's just for readability purposes